Config/CommonConfiguration.swift

```diff
@@ -82,6 +91,64 @@ class CommonConfiguration: NSObject, Configurable {
         MXKeyProvider.sharedInstance().delegate = EncryptionKeyManager.shared
     }
     
+    private func makeASCIIUserAgent() -> String? {
+        guard var userAgent = makeUserAgent() else {
+            return nil
+        }
+        if !userAgent.canBeConverted(to: .ascii) {
+            let mutableUserAgent = NSMutableString(string: userAgent)
+            if CFStringTransform(mutableUserAgent, nil, "Any-Latin; Latin-ASCII; [:^ASCII:] Remove" as CFString, false) {
+                userAgent = mutableUserAgent as String
+            }
+        }
+        return userAgent
+    }
+    
+    private func makeUserAgent() -> String? {
+        let clientName = Bundle.main.object(forInfoDictionaryKey: kCFBundleExecutableKey as String) as? String ?? Bundle.main.object(forInfoDictionaryKey: kCFBundleIdentifierKey as String) as? String ?? "unknown"
+        let clientVersion = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? Bundle.main.object(forInfoDictionaryKey: kCFBundleVersionKey as String) as? String ?? "unknown"
+
+    #if os(iOS)
+        return String(
+            format: "%@/%@ (%@; iOS %@; Scale/%0.2f)",
+                clientName,
+                clientVersion,
+                Device.current.safeDescription,
+                UIDevice.current.systemVersion,
+                UIScreen.main.scale)
+    #else
+    #if os(tvOS)
+        return String(
+            format: "%@/%@ (%@; tvOS %@; Scale/%0.2f)",
+                clientName,
+                clientVersion,
+                Device.current.safeDescription,
+                UIDevice.current.systemVersion,
+                UIScreen.main.scale)
+    #else
+    #if os(watchOS)
+        return String(
+            format: "%@/%@ (%@; watchOS %@; Scale/%0.2f)",
+                clientName,
+                clientVersion,
+                Device.current.safeDescription,
+                WKInterfaceDevice.current.systemVersion,
+                WKInterfaceDevice.currentDevice.screenScale)
+    #else
+    #if os(OSX)
+        return String(
+            format: "%@/%@ (Mac; Mac OS X %@)",
+                clientName,
+                clientVersion,
+                NSProcessInfo.processInfo.operatingSystemVersionString)
+    #else
+        return nil
+    #endif
+    #endif
+    #endif
+    #endif
```

Comment:
This is a bit annoying but Swift doesn't seem to have an `#elseif`. This is the only way I found to avoid the compiler warning for an unused `return nil`.